stellarphot/core.py

```diff
@@ -565,35 +579,21 @@ class PhotometryData(BaseEnhancedTable):
         "file": None,
     }
     observatory = None
-    camera = None
+    camera = TableAttribute(default=None)
 
     def __init__(
         self,
         *args,
         input_data=None,
         observatory=None,
-        camera=None,
         colname_map=None,
         passband_map=None,
         retain_user_computed=False,
         **kwargs,
     ):
-        if (observatory is None) and (camera is None) and (input_data is None):
+        if (observatory is None) and (self.camera is None) and (input_data is None):
```

Comment:
Wait, self.camera is a TableAttribute... but it defaults to None, so I assume this means it reports its value as None.  I assume you checked this.

Reply:
I added a couple of lines to one of the tests to check this. As it turns out, the old implementation didn't work when `camera` or `observatory`  were `None` because accessing `phot_table.camera` returned the `property`, which tries to construct a `Camera` even if the there wasn't one when the object was created.